src/TabBar.tsx

```diff
@@ -264,19 +264,17 @@ export default class TabBar<T extends Route> extends React.Component<
   private handleLayout = (e: LayoutChangeEvent) => {
     const { height, width } = e.nativeEvent.layout;
 
-    if (
-      this.state.layout.width === width &&
-      this.state.layout.height === height
-    ) {
-      return;
-    }
-
-    this.setState({
-      layout: {
-        height,
-        width,
-      },
-    });
+    // Decided not to place layout in state because
+    // onLayout is often called before applying changes
+    // (according to https://facebook.github.io/react-native/docs/view.html#onlayout)
+    // and new state (and related evaluation)
+    // can be called before actually applying new related which might lead
+    // to unexpected position of components.
+    // Furthermore, component gets updated anyway after changing orientation
```

Comment:
I think this is not sufficient. TabBar's layout can change for a variety of reasons, not only orientation, e.g. stuff like margin or width of tab bar changing with `style` prop.

Does adding `this.forceUpdate()` after this seem reasonable?